src/main/java/org/opensearch/ad/InsightsJobProcessor.java

```diff
@@ -0,0 +1,621 @@
+                                    lock -> runInsightsJob(jobParameter, lockService, lock, executionStartTime, executionEndTime),
+                                    exception -> {
+                                        log
+                                            .error(
+                                                new ParameterizedMessage("Failed to acquire lock for Insights job {}", jobName),
+                                                exception
+                                            );
+                                        // No lock to release on acquisition failure
+                                    }
+                                )
+                        );
+                } else {
+                    log.warn("No lock duration configured for Insights job: {}", jobName);
+                }
+            } catch (Exception e) {
+                log.error(new ParameterizedMessage("Error executing Insights job {}", jobName), e);
+            }
+        };
+
+        localThreadPool.executor(localThreadPoolName).submit(runnable);
+    }
+
+    /**
+     * Release lock for job.
+     */
+    private void releaseLock(Job jobParameter, LockService lockService, LockModel lock) {
+        lockService
+            .release(
+                lock,
+                ActionListener.wrap(released -> { log.info("Released lock for Insights job {}", jobParameter.getName()); }, exception -> {
+                    log.error(new ParameterizedMessage("Failed to release lock for Insights job {}", jobParameter.getName()), exception);
+                })
+            );
+    }
+
+    /**
+     * Execute the main Insights job logic.
+     * 
+     * @param jobParameter The insights job
+     * @param lockService Lock service for releasing lock
+     * @param lock The acquired lock
+     * @param executionStartTime Start of analysis window
+     * @param executionEndTime End of analysis window
+     */
+    private void runInsightsJob(
+        Job jobParameter,
+        LockService lockService,
+        LockModel lock,
+        Instant executionStartTime,
+        Instant executionEndTime
+    ) {
+        String jobName = jobParameter.getName();
+        if (lock == null) {
+            log.warn("Can't run Insights job due to null lock for {}", jobName);
+            return;
+        }
+
+        log.info("Running Insights job for time window: {} to {}", executionStartTime, executionEndTime);
+
+        querySystemResultIndex(jobParameter, lockService, lock, executionStartTime, executionEndTime);
+    }
+
+    /**
+     * Query all anomalies from system result index for the given time window.
+     * 
+     * @param jobParameter The insights job
+     * @param lockService Lock service for releasing lock
+     * @param lock The acquired lock
+     * @param executionStartTime Start of analysis window
+     * @param executionEndTime End of analysis window
+     */
+    private void querySystemResultIndex(
```

Comment:
Why do we limit ourselves to system index?

Reply:
we only query on auto created detectors, which use system result index (in the current iteration)